images/operator/Dockerfile

```diff
@@ -58,6 +58,10 @@ RUN --mount=type=bind,readwrite,target=/go/src/github.com/cilium/cilium \
     ./build-gops.sh
 
 FROM ${BASE_IMAGE}
+
+# Datadog Modification: operator has to run as root
+USER root
```

Comment:
🤔 I wonder if we actually need this now days, I suspect not.

Reply:
Actually we do... I believe this might also be somehow related to the fact that we use GBI images as base? This is what happened when I tried to run 1.12 without root: [logs](https://ddstaging.datadoghq.com/logs?query=index%3Ak8s-platform%20kube_cluster_name%3Agenesect%20service%3Acilium-operator%20image_tag%3Av18446891-74e3cba6-1.12.11%20&cols=datacenter%2Cservice&event=AgAAAYmw8yviGUWWsQAAAAAAAAAYAAAAAEFZbXc4ekRPQUFELVdJTi03RUstdVFBQgAAACQAAAAAMDE4OWIwZjctN2FmNi00ODExLTg2N2EtNGZhYThlOGI4Nzc0&index=k8s-platform&messageDisplay=inline&stream_sort=desc&viz=stream&from_ts=1690840227290&to_ts=1690906191362&live=false)